### PR TITLE
Adds TorrentRepoImpl, refactors handshake listener.

### PR DESCRIPTION
**Pull request by Lorraine-Lyu**
@JoshuaEng I changed the code in OwlTorrentClient to start the listener. It seems that the Client will also be refactored for MVP.

### CODE REVIEW

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerConnector.java [hunk 1]
```diff
@@ -21,7 +21,7 @@
   protected final Peer peer;
   // TODO: bad practice, should eventually refactor
   protected final TorrentManager manager;
-  protected final StorageAdapter storageAdapter;
+  protected StorageAdapter storageAdapter;
```
Comment (Lorraine-Lyu) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerConnector.java:24]: ClientHandler gets new peerConnector and store it to TorrentManage.

ClientHandler does not have access to storageAdapter do this field needs to be set later.

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/ClientHandler.java [hunk 2]
```diff
@@ -34,23 +31,27 @@ public void run() {
       in = new DataInputStream(socket.getInputStream());
 
       byte[] handShakeBuffer = new byte[68];
-      // TODO: handle read != 68 bytes
-      in.read(handShakeBuffer);
-      Optional<Peer> peer = verifyHandShake(handShakeBuffer);
-      if (peer.isEmpty()) {
+      if (in.read(handShakeBuffer) != 68) {
+        log.info("Received invalid handshake.");
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/ClientHandler.java:35]: in the case of invalid handshake, probably want to log error, and also return instead.

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/ClientHandler.java [hunk 3]
```diff
@@ -34,23 +31,27 @@ public void run() {
...
+      }
+      Optional<TorrentManager> torrentManager = verifyHandShake(handShakeBuffer);
+      if (torrentManager.isEmpty()) {
         return;
       }
+
       // TODO: fill in NetworkToStorage adapter and messageReader.
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/ClientHandler.java:42]: delete this TODO. (also I like this way of refactoring)

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerConnector.java [hunk 4]
```diff
@@ -38,6 +38,10 @@
    */
   protected abstract void respondToConnection() throws IOException;
 
+  void setStorageAdapter(StorageAdapter adapter) {
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/PeerConnector.java:41]: consider using the @Setter annotation with access level = Package Private

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java [hunk 5]
```diff
@@ -50,7 +50,15 @@ private SocketConnector(
       StorageAdapter storageAdapter,
       MessageReader messageReader,
       Socket peerSocket) {
-    super(peer, manager, storageAdapter, messageReader);
+    super(peer, manager, messageReader);
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java:53]: maybe change this to this(peer, manager, messageReader, peerSocket), and then delete line 55?

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.entity.TwentyByteId;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "general")
+public class TorrentRepositoryImpl implements TorrentRepository {
+  private ConcurrentHashMap<TwentyByteId, TorrentManager> infoHashToTorrentManager;
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java:10]: make this final. nit: Also can inline the call to new and have an empty constructor.

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
+package edu.rice.owltorrent.network;
+
+import edu.rice.owltorrent.common.entity.TwentyByteId;
+import java.util.Optional;
+import java.util.concurrent.ConcurrentHashMap;
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2(topic = "general")
+public class TorrentRepositoryImpl implements TorrentRepository {
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java:9]: add javadoc

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+
+  @Override
+  public void registerTorrentManager(TorrentManager manager) {
+    TwentyByteId infoHash = manager.getTorrent().getInfoHash();
+    /* This conditional should not be triggered in any case. */
+    if (infoHashToTorrentManager.containsKey(infoHash)) {
+      log.error("Torrent manager for torrent {} already exist.", manager.getTorrent().getName());
+      return;
+    }
+    this.infoHashToTorrentManager.put(infoHash, manager);
```
Comment (maxyu1115) [inline owltorrent-network/src/main/java/edu/rice/owltorrent/network/TorrentRepositoryImpl.java:24]: This shouldn't matter much, but checking if contains or not and then put is not thread safe. You want to use something like putIfAbsent instead.

File: owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java (hunk 5 above)
Comment (Lorraine-Lyu) [reply owltorrent-network/src/main/java/edu/rice/owltorrent/network/SocketConnector.java:53]: Socket should not be a field in PeerConnector, right?

**Review by maxyu1115 — APPROVED** [review]

**Pull request merged**